common/src/main/java/com/wynntils/features/combat/RangeVisualizerFeature.java

```diff
@@ -14,91 +14,132 @@
+        // since the previous tick
+        circlesToRender.clear();
+        detectedPlayers.forEach(this::checkMajorIdCircles);
+        detectedPlayers.clear();
     }
 
-    private CustomColor getCircleColor(Player player) {
-        if (!Models.Player.isLocalPlayer(player)) return null; // Don't render for ghost/npc
-        String playerName = ComponentUtils.getUnformatted(player.getName());
-        boolean isSelf =
-                ComponentUtils.getUnformatted(McUtils.player().getName()).equals(playerName);
-        if (isSelf && McUtils.mc().screen instanceof InventoryScreen)
-            return null; // Don't render for preview in inventory
-        if (!Models.Party.getPartyMembers().contains(playerName) && !isSelf)
-            return null; // Other players must be in party
-
-        // We are getting the item info the same way as GearViewerScreen since we care about other people's items
-        String gearName = ComponentUtils.getUnformatted(player.getMainHandItem().getHoverName());
-        GearInfo gearInfo = Models.Gear.getGearInfoFromApiName(gearName);
-        if (gearInfo == null) return null;
-
-        if (isSelf) { // Do not render if the item is not for the player's class
-            if (gearInfo.requirements().classType().isEmpty()) return null;
-            ClassType classType = gearInfo.requirements().classType().get();
-            if (classType != Models.Character.getClassType()) return null;
+    private void checkMajorIdCircles(Player player) {
+        // Don't render for ghost/npc
+        if (!Models.Player.isLocalPlayer(player)) return;
+
+        List<GearInfo> validGear;
+
+        if (player == McUtils.player()) {
+            // This is ourselves, rendered from outside
+
+            // Don't render for preview in inventory or character selection screen
+            if (McUtils.mc().screen instanceof InventoryScreen) return;
+            if (McUtils.mc().screen instanceof CharacterSelectorScreen) return;
+
+            validGear = Models.CharacterStats.getWornGear();
+        } else {
+            // Other players must be in party
+            if (!Models.Party.getPartyMembers().contains(ComponentUtils.getUnformatted(player.getName()))) return;
+
+            validGear = new ArrayList<>();
+            // Check main hand
+            GearInfo mainHandGearInfo = getOtherPlayerGearInfo(player.getMainHandItem());
+            if (mainHandGearInfo != null) {
+                if (mainHandGearInfo.type().isWeapon()) {
+                    // We cannot verify class or level :(
+                    validGear.add(mainHandGearInfo);
+                }
+            }
+
+            // Check armor slots
+            player.getArmorSlots().forEach(itemStack -> {
+                GearInfo armorGearInfo = getOtherPlayerGearInfo(player.getMainHandItem());
+                if (armorGearInfo != null) {
+                    if (armorGearInfo.type().isArmour()) {
```

Comment:
`isArmour`? I thought we used `isArmor`, the American english version. But since this is not added in this PR, no need to change.

Reply:
Yes, I reacted to that, too. We need to fix it in a separate PR.

Reply:
Ah, what the heck. I fixed it in this.